packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -222,208 +226,140 @@ library LibProving {
-            // but immediately validates it, obviating the requirement to set a
-            // contester, burn the contest bond, and other associated actions.
-            // This streamlined process is applied to 0-to-non-zero transition
-            // updates.
-            if (
-                ts.contester == address(0) && ts.blockHash == tran.blockHash
-                    && ts.signalRoot == tran.signalRoot
-            ) {
-                // Alternatively, it can be understood that a transition cannot
-                // be re-approved by higher-tier proofs without undergoing
-                // contestation.
-                revert L1_ALREADY_PROVED();
-            }
+            if (tier.contestBond == 0) {
+                // On the highest tier
+                assert(tier.validityBond == 0);
+                assert(ts.validityBond == 0 && ts.contestBond == 0 && ts.contester == address(0));
 
-            if (tid == 1 && ts.tier == 0 && block.timestamp <= ts.timestamp + tier.provingWindow) {
-                // For the first transition, (1) if the previous prover is
-                // still the assigned prover, we exclusively grant permission to
-                // the assigned approver to re-prove the block, (2) unless the
-                // proof window has elapsed.
-                if (msg.sender != blk.assignedProver) revert L1_NOT_ASSIGNED_PROVER();
-            } else if (msg.sender == blk.assignedProver) {
-                // However, if the previous prover of the first transition is
-                // not the block's assigned prover, or for any other
-                // transitions, the assigned prover is not permitted to prove
-                // such transitions.
-                revert L1_ASSIGNED_PROVER_NOT_ALLOWED();
-            }
+                ts.prover = msg.sender;
+                ts.blockHash = tran.blockHash;
+                ts.signalRoot = tran.signalRoot;
 
-            unchecked {
-                // This is the amount of Taiko tokens to send to the new prover
-                // and the winner of the contest (same amount to both parties).
-                uint256 reward;
-                if (ts.blockHash == tran.blockHash && ts.signalRoot == tran.signalRoot) {
-                    assert(ts.contester != address(0));
-                    // In the event that the previous prover emerges as the
-                    // winner, half of the contest bond is designated as the
-                    // reward, to be divided equally between the new prover and
-                    // the previous prover -- 1/4 each
-                    reward = ts.contestBond >> 2;
+                emit TransitionProved({
+                    blockId: blk.blockId,
+                    tran: tran,
+                    prover: msg.sender,
+                    validityBond: 0,
+                    tier: proof.tier
+                });
+            } else {
```

Comment:
According to this new logic, there shall be an extra check (but please confirm) somewhere here. 

Because this [line](https://github.com/taikoxyz/taiko-mono/blob/29e82230d09080b263c6aabb937550c9fa7a63ca/packages/protocol/contracts/L1/libs/LibProving.sol#L243) will allow people to contest with lower tiers !

Example, please follow along:

So tiers are: 1.Optimistic, 2. Sgx, 3. ZK+Sgx, 4. Guardian

1. Bob proves a Zk+SGX block (because the default tier is 3).
2. Alice wants to contest the block and she submits a proof with SGX , and she will succeed to contest.